.github/workflows/pod_failover_test.yml

```diff
@@ -0,0 +1,81 @@
+name: Pod Failover Test
+on:
+    schedule:
+    - cron: "0 13 * * *"
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    env:
+      AZURE_TENANT_ID: ${{ secrets.TenantId }}
+      AZURE_SUBSCRIPTION_ID: ${{ secrets.SubscriptionId }}
+      AZURE_CLIENT_ID: ${{ secrets.ClientId }}
+      AZURE_CLIENT_SECRET: ${{ secrets.ClientSecret }}
+      AZURE_LOCATION: ${{ secrets.E2ELocation }}
+      REGISTRY: e2etest.azurecr.io/mandalay
+      METRICS_ENDPOINT: ${{ secrets.MetricsEndpoint }}
+    
+    steps:
+    - name: Set UUID
+      id: generate-uuid
+      uses: filipstefansson/uuid-action@v1
+
+    - name: Set Azure resource group
+      run: |
+        echo "AZURE_RESOURCE_GROUP=k8s-e2eTest-${{ steps.generate-uuid.outputs.uuid }}-rg" >> $GITHUB_ENV
+ 
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: ^1.16
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Install conntrack
+      run: sudo apt install conntrack
+
+    - name: Login to Azure
+      run: |
+        az login --service-principal -u $AZURE_CLIENT_ID -p $AZURE_CLIENT_SECRET --tenant $AZURE_TENANT_ID
+
+    - name: Login to aks registry
+      run: az acr login --name e2etest
+
+    - name: Build Test
+      env:
+        BUILD_V2: true
+      run: |
+        make container-all push-manifest
+
+    - name: Create an aks cluster 
+      run: |  
+        ./hack/azure-cluster-up.sh --subscription $AZURE_SUBSCRIPTION_ID --location $AZURE_LOCATION --output e2eTestOutput --client-id $AZURE_CLIENT_ID --client-tenant $AZURE_TENANT_ID --client-secret $AZURE_CLIENT_SECRET --resource-group $AZURE_RESOURCE_GROUP
+        echo "KUBECONFIG=$GITHUB_WORKSPACE/e2eTestOutput/kubeconfig/kubeconfig.${AZURE_LOCATION}.json" >> $GITHUB_ENV
+
+    - name: Deploy the driver on the cluster
+      env:
+        BUILD_V2: true
+      run: make e2e-bootstrap
+
+    - name: Build controller and workload containers
+      run: make pod-failover-test-containers
+
+    - name: Get the deployment file and modify it
+      run: |
+        wget https://raw.githubusercontent.com/abhisheksinghbaghel/azuredisk-csi-driver/master/test/podFailover/deployment.yaml
```

Comment:
It would be ideal to run multiple failover test cases, e.g. 1pod, 1vol; 1pod, 3vol; 3 pods on same node, 1 vol. Could this be done in a single action using Kubernetes jobs? The challenge, I think, would be in getting the start and end of each job to know the time period in which to gather the metrics for the dashboard. Could we generate links to the Grafana dashboard with the appropriate time ranges after the run?

Reply:
We can run multiple test cases as part of this workflow. I've added 2 tests for now- 1pod,1vol and 1 pod,3vol. Also introduced a metrics label called testName to differentiate between the metrics generated from the different test runs.

I'm not sure if we need to track the time ranges of the test runs. I'm thinking of creating grafana dashboards for different test runs based on their testName labels. The refresh time of the dashboards can be set to 24 hours, so that every dashboard refreshes once everyday. We could also configure the dashboard to only collect metrics from last 24 hours. Since(I assume), we will be running these tests only once per day, every dashboard refresh will show the data from the last test run. 

I hope that makes sense? Let me know if you think there can be problems with this approach.